linera-views/src/reentrant_collection_view.rs

```diff
@@ -210,7 +211,8 @@ where
         Ok(ReentrantByteCollectionView {
             context: self.context.clone(),
             delete_storage_first: self.delete_storage_first,
-            updates: Mutex::new(cloned_updates),
+            updates: cloned_updates,
+            cached_entries: Mutex::new(BTreeMap::new()),
```

Comment:
It's safe, but I think there's a risk the overhead is more than expected. That's because we'd have to do the same we did for the `updates`, locking every entry's write-lock and calling `clone_unchecked`.

There's also a risk that it ends up unexpectedly changed by a bad implementation of `clone_unchecked(&mut self)` :(